Fix vertical scrolling on IE when the table auto-detects its height

adube opened the pull request:
### Context

I would like to restore an old patch that was provided by @samuellapointe back in 2015.  Here's the original context of the original issue (#2942):

> The issue (#2924) was in a project where I had to let a table fill a container, it didn't create scrollbars on IE. It's related to how the table selects its height according to the computed style. The previous behavior first checked if element.currentStyle returned a style, otherwise it used document.defaultView.getComputedStyle. However, element.currentStyle returns incorrect width and height values: auto. Since IE9+ is compatible with document.defaultView.getComputedStyle, I set it to first check if that function exists, and if so, use it. Otherwise, use element.currentStyle. This means the bug is fixed on IE9+ but it will still happen on previous versions.

Context of the original PR (#2663)

>This commit fixes the issue #2942 where using a hook on afterSelectionEnd didn't work on mobile. To do that, I added an event listener to the touchend event, and make it end a selection if there's one. To make sure it is fired after the selection occurs, it's placed in a setTimeout of length 0. Otherwise it would sometimes fire before the selection started.

### How has this been tested?
See in the description above.

### Types of changes
- [x] Bug fix (non-breaking change which fixes an issue)

### Related issue(s):
1. #2942
2. #2924 
3. #2963 (PR)

### Checklist:
- [x] My code follows the code style of this project (I'm pretty sure it does)

adube commented:
Please, let me know if I missed anything.

AMBudnik commented:
Hi @adube 

I see that the branch is correct and the CLA is signed but we would also need some tests.

adube commented:
Hello @AMBudnik.  Thanks for the quick reply.

How's this test?

AMBudnik commented:
Thank you for adding the test @adube 

Yes, we do not merge PRs that change anything in the `dist` directory (point `4` in [the Guide](https://github.com/handsontable/handsontable/blob/master/CONTRIBUTING.md))

adube commented:
@AMBudnik Thanks for pointing this out.  I updated the patch accordingly and rebased onto the up-to-date develop branch.  I think this is ready for review.

Please, let me know if I need to do anything else.

AMBudnik commented:
That's correct @adube 
I'm adding the label.

adube commented:
Rebased onto "develop".

pnowak commented:
Hi @adube 

Sorry for bringing the bad news but I'm not sure that more verbose checking fixed issue. I can't test it because of the merge conflicts. Moreover, we don't use `document` directly in our code.

 So we need to close this PR.

Thank you very much for sharing it and doing the investigation.

adube commented:
No worries.  Thanks for the feedback.
